### PR TITLE
Fibonacci_Number using DP approach added

### LINKED ISSUE #1948: Fibonacci using DP Approach
There is no issue present on Fibonacci. i would to implement the fibonacci using Dynamic Programming Approach. please assign this to me. 
@SakshiShreya @asha15 @jainaman224 @MastersAbh @Vaibhavraj10

### PR DESCRIPTION
**Pull request by ravireddy07**
Fixes #1948 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Fibonacci using Dynamic Programming approach

#### Languages Used:
- C
- C++
- Java

#### Files Added:
- Fibonacci_NumberDP.c
- Fibonacci_NumberDP.cpp
- Fibonacci_NumberDP.java

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by ravireddy07** [comment]
@somya-kapoor  I've added only three files. I don't know from where all this commits and files came to this PR.

**Comment by somya-kapoor** [comment]
@ravireddy07 Yes, please delete them.

**Comment by ravireddy07** [comment]
@somya-kapoor I've removed all otehr files whuch are not related to this PR and squashed all commits to one. Please review it once.

File: Fibonacci_Number/Fibonacci_NumberDP.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.c]: remove this line

File: Fibonacci_Number/Fibonacci_NumberDP.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+int dp[100005] = {0};
+
+int fibDP(int n)
+{
+    if(n==0 || n==1)
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.c]: keep space around ==

File: Fibonacci_Number/Fibonacci_NumberDP.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        return n;
+    }
+    if(dp[n] != 0)
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.c]: it's better if you explain this in a comment

File: Fibonacci_Number/Fibonacci_NumberDP.c [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        return dp[n];
+    }
+    int cand1 = fibDP(n-1);
+    int cand2 = fibDP(n-2);
+
+    dp[n] = cand1+cand2;
+    return cand1+cand2;
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.c]: keep space around operator

File: Fibonacci_Number/Fibonacci_NumberDP.c [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+int main(int argc, char const *argv[])
+{
+    int n, result;
+    printf("Enter a Number to find the Fibonacci using DP apporach: ");
+    scanf("%d",&n);
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.c]: keep a space after comma

File: Fibonacci_Number/Fibonacci_NumberDP.cpp [hunk 6]
```diff
@@ -0,0 +1,41 @@
+#include <iostream>
+using namespace std;
+
+int dp[100005] = {0};
+
+int fibDP(int n)
+{
+    if(n==0 or n==1)
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.cpp]: keep space around operator

File: Fibonacci_Number/Fibonacci_NumberDP.c [hunk 7]
```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+
+int dp[100005] = {0};
+
+int fibDP(int n)
+{
+    if(n==0 || n==1)
+    {
+        return n;
+    }
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.c]: keep a line space

File: Fibonacci_Number/Fibonacci_NumberDP.cpp [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        return n;
+    }
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.cpp]: keep a line space

File: Fibonacci_Number/Fibonacci_NumberDP.cpp [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    if(dp[n] != 0)
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.cpp]: Explain this in a comment

File: Fibonacci_Number/Fibonacci_NumberDP.cpp [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        return dp[n];
+    }
+    int cand1 = fibDP(n-1);
+    int cand2 = fibDP(n-2);
+
+    dp[n] = cand1+cand2;
+    return cand1+cand2;
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.cpp]: keep space around operator

File: Fibonacci_Number/Fibonacci_NumberDP.cpp [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+int main(int argc, char const *argv[])
+{
+    int n;
+    cout<<"Enter a Number to find the Fibonacci using DP apporach: ";
+    cin>>n;
+    cout<<"Fibonacci of "<<n<<" is: "<<fibDP(n);
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.cpp]: keep space around << and >>

File: Fibonacci_Number/Fibonacci_NumberDP.java [hunk 12]
```diff
@@ -0,0 +1,47 @@
+import java.util.*;
+
+class findingFib
+{
+    int dp[] = new int[100005];
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.java]: keep a line space

File: Fibonacci_Number/Fibonacci_NumberDP.java [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+    int fibDP(int n)
+    {
+        if(n==0 || n==1)
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.java]: keep space around ==

File: Fibonacci_Number/Fibonacci_NumberDP.java [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+	{
+	    return n;
+	}
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.java]: keep a line space

File: Fibonacci_Number/Fibonacci_NumberDP.java [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+        if(dp[n] != 0)
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.java]: add a comment

File: Fibonacci_Number/Fibonacci_NumberDP.java [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+	{
+	    return dp[n];
+	}
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.java]: keep a line space here

File: Fibonacci_Number/Fibonacci_NumberDP.java [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+	int cand1 = fibDP(n-1);
+	int cand2 = fibDP(n-2);
+	    
+	dp[n] = cand1+cand2;
+	return cand1+cand2;
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.java]: keep space around operator

File: Fibonacci_Number/Fibonacci_NumberDP.java [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+{
+    public static void main(Strinxg args[])
+    {
+        findingFib ob1 = new findingFib();
+	int n, result;
+	Scanner s = new Scanner(System.in);
+	System.out.print("Enter a Number to find the Fibonacci using DP apporach: ");
+	n = s.nextInt();
+	result = ob1.fibDP(n);
+	System.out.println("Fibonacci of "+n+" is: "+result);
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.java]: please correct the indentation

**Comment by ravireddy07** [comment]
@asha15  @somya-kapoor  required changes has been pushed, once check.

File: Fibonacci_Number/Fibonacci_NumberDP.c [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        return dp[n];
+    }*/
+    
+    /* This recursion is to hold the values till cand1 = 1(from 5-4-3-2-1), cand2 = 1(5-3-1) */
+    int cand1 = fibDP(n - 1);
+    int cand2 = fibDP(n - 2);
+    /* Later, it will go back to its previous values(2-3), cand1, cand2 values will be updated.
+    /* after, getting values (1,1), those values get added and updated at 5th index. this will repeat in each iteration*/
+    /* you can check(print) manually the values of cand1, cand2, n and dp[n] at multiple instances to get a proper idea of structure of the program */
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.c]: make this as a multi line comment
one line should contain only 80 characters 
please changes this in all the other files too

Comment (ravireddy07) [reply Fibonacci_Number/Fibonacci_NumberDP.c]: Yes, changed it to multi-line comment. please chack it.

File: Fibonacci_Number/Fibonacci_NumberDP.c [hunk 20]
```diff
@@ -7,22 +7,23 @@ int fibDP(int n)
...
     /* This recursion is to hold the values till cand1 = 1(from 5-4-3-2-1), cand2 = 1(5-3-1) */
     int cand1 = fibDP(n - 1);
     int cand2 = fibDP(n - 2);
-    /* Later, it will go back to its previous values(2-3), cand1, cand2 values will be updated.
-    /* after, getting values (1,1), those values get added and updated at 5th index. this will repeat in each iteration*/
-    /* you can check(print) manually the values of cand1, cand2, n and dp[n] at multiple instances to get a proper idea of structure of the program */
+    /* 
+	Later, it will go back to its previous values(2-3), cand1, cand2 values will be updated.
+    after, getting values (1,1), those values get added and updated at 5th index. this will repeat in each iteration,
+    You can check(print) manually the values of cand1, cand2, n and dp[n] at multiple instances to get a proper idea of structure of the program
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.c]: break this into another line

File: Fibonacci_Number/Fibonacci_NumberDP.cpp [hunk 21]
```diff
@@ -9,22 +9,23 @@ int fibDP(int n)
...
     /* This recursion is to hold the values till cand1 = 1(from 5-4-3-2-1), cand2 = 1(5-3-1) */
     int cand1 = fibDP(n - 1);
     int cand2 = fibDP(n - 2);
-    /* Later, it will go back to its previous values(2-3), cand1, cand2 values will be updated.
-    /* after, getting values (1,1), those values get added and updated at 5th index. this will repeat in each iteration*/
-    /* you can check(print) manually the values of cand1, cand2, n and dp[n] at multiple instances to get a proper idea of structure of the program */
+    /* 
+	Later, it will go back to its previous values(2-3), cand1, cand2 values will be updated.
+    after, getting values (1,1), those values get added and updated at 5th index. this will repeat in each iteration,
+    you can check(print) manually the values of cand1, cand2, n and dp[n] at multiple instances to get a proper idea of structure of the program.
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.cpp]: break this into another line

File: Fibonacci_Number/Fibonacci_NumberDP.java [hunk 22]
```diff
@@ -11,22 +11,22 @@ int fibDP(int n)
...
 	
 	/* This recursion is to hold the values till cand1 = 1(from 5-4-3-2-1), cand2 = 1(5-3-1) */
 	int cand1 = fibDP(n-1);
 	int cand2 = fibDP(n-2);
-	/* Later, it will go back to its previous values(2-3), cand1, cand2 values will be updated. */
-	/* after, getting values (1,1), those values get added and updated at 5th index. this will repeat in each iteration */
-	/* you can check(print) manually the values of cand1, cand2, n and dp[n] at multiple instances to get a proper idea of structure of the program */
+	/* Later, it will go back to its previous values(2-3), cand1, cand2 values will be updated.
+	   after, getting values (1,1), those values get added and updated at 5th index. this will repeat in each iteration,
+	   you can check(print) manually the values of cand1, cand2, n and dp[n] at multiple instances to get a proper idea of structure of the program.
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.java]: break the into another line

Comment (ravireddy07) [reply Fibonacci_Number/Fibonacci_NumberDP.java]: breaked into another line... Please check now.

File: Fibonacci_Number/Fibonacci_NumberDP.cpp [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        return dp[n];
+    }*/
+    
+    /* This recursion is to hold the values till cand1 = 1(from 5-4-3-2-1), cand2 = 1(5-3-1) */
+    int cand1 = fibDP(n - 1);
+    int cand2 = fibDP(n - 2);
+    /* 
+	Later, it will go back to its previous values(2-3), cand1, cand2 values will be updated.
+    after, getting values (1,1), those values get added and updated at 5th index.
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.cpp]: you should keep another space at the beginning

File: Fibonacci_Number/Fibonacci_NumberDP.cpp [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+	this will repeat in each iteration,
+	
+	You can check(print) manually the values of cand1, cand2, n and dp[n] 
+	at multiple instances to get a proper idea of structure of the program
+   
+    dp[n] = cand1 + cand2; This statement is used to store the fibonacci series in the array
+    
+    here, we will be returning the final values of cand1, cand2. */
```
Comment (asha15) [inline Fibonacci_Number/Fibonacci_NumberDP.cpp]: remove the extra line
correct this in all the files and let me know

**Comment by somya-kapoor** [comment]
@ravireddy07 Are you working on it?

**Comment by ravireddy07** [comment]
Yes. I've updated the code. Corrected the requested indentation. Please check.
@somya-kapoor  @asha15

**Comment by ravireddy07** [comment]
@asha15  renamed the file names according to it and copy to new folder. Please check it.

**Comment by Manish-cloud** [comment]
@asha15 I would like to provide fibonacci number implementation using DP based approach in python language

**Pull request merged**